wiki/PreventiveMaintenance.md

```diff
@@ -0,0 +1,64 @@
+
+## How?
+Keep in mind that documentation is different than comments. Comments are placed between code lines in the source file, whereas documentation is placed in the header file before function and class definitions. Good code should be self-explanatory which means that it often does not need any comments, when properly split into smaller functions that have been well-documented. And when writing comments, it should mostly be used to answer the why questions rather than the what questions, i.e. comments should focus on why does this part of code works, why don't I have to check certain situations, etc. rather than what the code actually does. Documentation on the other hand can be used to explain what the code does, but should also be used to explain why this code works. Documentation should give an answer to the most important questions regarding the code which can include answers on questions like:
+* What does the code compute? What is the meaning of the output? What are the postconditions on the output? What is the meaning of the input arguments? Which preconditions should hold on the input arguments? Which requirements should be satisfied before using the code? Etc.
+* Is there some known bug/strange behavior in the code which hasn't been solved yet? How does the code respond in case of an exception? What happens when preconditions/requirements have been violated? Etc.
+* What is the meaning of abbreviations used inside the code? Is there a website which explains (partially) the meaning of the code? From which source has this code been copied? Etc.
+* In case of class documentation: Who has written/maintained this code? At which date has this code been written?
+* Less important, but could be included as well: Why does this code work? Why is this code efficient? Why did you choose this type of design? What other solutions have you tried out for this code? What major decisions have been made regarding this code? 
+
+## Good Example
+Below is an example how you should document your code:
+```
+/**
+ * Get the intersection between two (finite) LineSegment. Note that this function knows how to deal with LineSegments that are points and how to deal with
+ * parallel LineSegments. Moreover this function guarantees to return an intersection point if the LineSegments intersects. In case:
+ * - There is no intersection then std:nullopt is returned.
+ * - There is a single intersection point then that intersection point is returned.
+ * - There are multiple intersection points (infinitely many) which happens when the LineSegments overlap then it prefers to returns 1: start, 2: end, 3: 
+ *  line.start of which the first most option is selected that lies on both LineSegments. Note that if none of these points lie on both LineSegments and if both 
+ *  the LineSegments are parallel then the LineSegments cannot intersect.
```

Comment:
I think this line also doensn't add information, you don't need to explain here what happens if there is no intersection. This information seems out of place here in my view
```suggestion
 * - There are multiple intersection points (infinitely many) which happens when the LineSegments overlap then it prefers to returns 1: start, 2: end, 3: 
 *  line.start of which the first most option is selected that lies on both LineSegments
```

Reply:
I guess you misread this. It explains what happens when there are multiple intersections which is quite important.

Reply:
I was talking about the part where it explains that if the lines are parallel then they don't intersect. This could be seen as relevant information, but I feel like if a paragraph gets too long people will start understanding it less easily. That's just my personal opinion, though. So I prefer to keep docs to say only what is absolutely necessary in order to understand the function. Of course, that's also a subjective measure. So there isn't really a right answer, I guess.

Reply:
But you are not changing anything related to that comment now...